Firmware/first_lay_cal.cpp

```diff
@@ -163,88 +178,93 @@ void lay1cal_before_meander()
 //! @brief Print meander start
 void lay1cal_meander_start(float layer_height, float extrusion_width)
 {
-    enquecommand_P(PSTR("G1 X50 Y155"));
-
-    static const char fmt1[] PROGMEM = "G1 Z%-.3f F7200";
+#ifndef NEW_FIRST_LAYER_CAL
+    enquecommand_P(PSTR("G1X50Y155"));
+#endif //_NEW_FIRST_LAYER_CAL
+    static const char fmt1[] PROGMEM = "G1Z%g";
     enquecommandf_P(fmt1, layer_height);
-
-    enquecommand_P(PSTR("G1 F1080"));
-
-    enquecommandf_P(extrude_fmt,  75, 155, count_e(layer_height, extrusion_width * 4.f, 25));
-    enquecommandf_P(extrude_fmt, 100, 155, count_e(layer_height, extrusion_width * 2.f, 25));
-    enquecommandf_P(extrude_fmt, 200, 155, count_e(layer_height, extrusion_width, 100));
-    enquecommandf_P(extrude_fmt, 200, 135, count_e(layer_height, extrusion_width, 20));
+    enquecommand_P(feedrate_F1080);
+#ifdef NEW_FIRST_LAYER_CAL
+    enquecommandf_P(extrude_fmt_Y, short_length, count_e(layer_height, extrusion_width, short_length));
+#endif //_NEW_FIRST_LAYER_CAL
+
+    enquecommand_P(MSG_G91); //enable relative XYZ
+#ifndef NEW_FIRST_LAYER_CAL
+    enquecommandf_P(extrude_fmt_X, (float)25*invert, count_e(layer_height, extrusion_width * 4.f, 25));
+    enquecommandf_P(extrude_fmt_X, (float)25*invert, count_e(layer_height, extrusion_width * 2.f, 25));
+    enquecommandf_P(extrude_fmt_X, (float)100*invert, count_e(layer_height, extrusion_width, 100));
+    enquecommandf_P(extrude_fmt_Y, (float)-20*invert, count_e(layer_height, extrusion_width, 20));
+#else
+    enquecommandf_P(extrude_fmt_X, long_length*invert, count_e(layer_height, extrusion_width, long_length));
```

Comment:
this line is overextruded since a shorter length is used with the same E count. Needs adjusting

![image](https://github.com/user-attachments/assets/19e5a2a5-9399-4342-a6a7-9bd7f8cf67d4)